File: java/src/main/java/algolens/twopointers/RemoveDuplicatesSorted.java
package algolens.twopointers;

public class RemoveDuplicatesSorted {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) return 0;
        int write = 1;
        for (int read = 1; read < nums.length; read++) {
            if (nums[read] != nums[write - 1]) {
                nums[write] = nums[read];
                write++;
            }
        }
        return write;
    }
}

File: javascript/src/two-pointers/remove-duplicates-sorted.js
import { parseAssignments } from "../shared/io.js";
export const implemented = true;
export function solve(input) {
  const { nums } = parseAssignments(input);
  if (nums.length === 0) return 0;
  let write = 1;
  for (let read = 1; read < nums.length; read++) {
    if (nums[read] !== nums[read - 1]) nums[write++] = nums[read];
  }
  return write;
}
